tldw-frontend/extension/components/Option/Collections/ReadingList/ReadingItemDetail.tsx

```
@@ -148,9 +198,18 @@ export const ReadingItemDetail: React.FC<ReadingItemDetailProps> = ({
   }, [highlightEditorOpen, currentItem?.id, fetchItemHighlights])
 
   const handleClose = useCallback(() => {
+    flushProgressSave()
     closeItemDetail()
     setEditingNotes(false)
-  }, [closeItemDetail])
+    setNotesDirty(false)
+    setNotesSaveError(null)
+    setNotesSaving(false)
+  }, [closeItemDetail, flushProgressSave])
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)